examplesV4/tokenCreationAndPaymentExample.php

```diff
@@ -0,0 +1,221 @@
+    ->withIdentityCardNumber('111222');
+
+/**
+ * Create new delivery address
+ *
+ * If you want to have the same delivery as billing, skip these two steps
+ * and pass the Billing $billing object to the request twice
+ */
+$delivery = new Delivery();
+
+/**
+ * Setup the delivery address params
+ *
+ * Full params available in the documentation
+ */
+$delivery->withAddressLine1('Address1')
+    ->withAddressLine2('Address2')
+    ->withCity('City')
+    ->withCountryCode('RO')
+    ->withEmail('john.doe@mail.com')
+    ->withFirstName('FirstName')
+    ->withLastName('LastName')
+    ->withPhoneNumber('40123456789');
+
+/**
+ * Create new Card with params:
+ *
+ * Credit Card Number
+ * Credit Card Expiration Month
+ * Credit Card Expiration Year
+ * Credit Card CVV (Security Code)
+ * Credit Card Owner
+ */
+$card = new Card(
+    '4111111111111111',
+    '01',
+    date("Y", strtotime("+1 year")),
+    123,
+    'Card Owner Name'
+);
+
+/**
+ * tokenize card for further token payments
+ */
+$card->enableTokenCreation();
+
+/**
+ * Create new Request with params:
+ *
+ * Config object
+ * Order object
+ * Billing object
+ * Delivery (or Billing object again, if you want to have the delivery address the same as the billing address)
+ * User object
+ */
+$request = new Request($cfg, $order, $billing, $delivery, $user, PaymentsV4::API_VERSION_V4);
+$requestWithToken = new Request($cfg, $order, $billing, $delivery, $user, PaymentsV4::API_VERSION_V4);
+$requestWithToken->getOrder()->withOrderRef('TokenOrderRef');
+/**
+ * Add the Credit Card to the Request
+ */
+$request->setCard($card);
+
+/**
+ * Create new API Client, passing the Config object as parameter
+ */
+$client = new Client($cfg);
+
+/**
+ * Will throw different Exceptions on errors
+ */
+try {
+    /**
+     * Sends the Request to ALU and returns a Response
+     *
+     * See documentation for Response params
+     */
+    $response = $client->pay($request);
+
+    /**
+     * In case of 3DS enrolled cards, PayU will return the URL_3DS that contains a unique url for each
+     * transaction. The merchant must redirect the browser to this url to allow user to authenticate.
+     * After the authentication process ends the user will be redirected to BACK_REF url
+     * with payment result in a HTTP POST request
+     */
+    if ($response->isThreeDs()) {
+        header("Location:" . $response->getThreeDsUrl());
+        die();
+    }
+
+    echo $response->getStatus() . ' ' . $response->getReturnCode() . ' ' . $response->getReturnMessage() . "\n";
+    echo('Token response data: ');
+    echo $response->getTokenCode() . ' ' . $response->getTokenMessage() . ' Token:' . $response->getTokenHash() . "\n";
+
+    if ($response->getTokenCode() == 0 && $response->getTokenHash() !== '') {
+        $cardToken = new CardToken($response->getTokenHash());
+        $requestWithToken->setCardToken($cardToken);
+
+        $response = $client->pay($requestWithToken);
```

Comment:
it is better not to reuse variables as it makes code more difficult to read and understand.
Maybe the first response could be called "initialPaymentResponse" and the second response "tokenPaymentResponse", or something else to make it clear what each represents

Reply:
done